adr/001-db.md

```diff
@@ -0,0 +1,68 @@
+# ADR-001: Choosing a database to store user subscriptions
+
+**Status:** Accepted
+
+**Date:** 2025-06-07
+
+**Author:** Stanislav Basarab
+
+## Context
+
+We need to store user subscriptions in a database. The database should be able to handle a large number of records, support complex queries, and provide good performance for read and write operations.
+
+## Considered options
+
+### 1. PostgreSQL
+
+**Pros:**
+- Stable and widely adopted open-source RDBMS.
+- Rich support for SQL features, indexing, and transactions (ACID compliance).
+- Well-supported by ORM libraries like Prisma, TypeORM, Sequelize.
+- Excellent integration with tools for backup, monitoring, and scaling.
+
+**Cons:**
+- Slightly more overhead in deployment/configuration compared to SQLite or Firebase.
+- Requires a separate running service (not embedded).
+
+---
+
+### 2. MongoDB
+
+**Pros:**
+- Schema-less, flexible data model.
+- Easy to start with, popular among JavaScript developers.
+- Scales horizontally more naturally for very large datasets.
+
+**Cons:**
+- Lacks strong relational features (joins are limited and expensive).
+- Risk of data inconsistency if not carefully modeled.
+
+---
+
+### 3. MySQL
+
+**Pros:**
+- Mature relational database with good performance and tooling.
+- ACID compliant and supports foreign keys, indexing, etc.
+- Broad support across hosting platforms and ORMs.
+- Slightly lower memory footprint than PostgreSQL in some configurations.
+
+**Cons:**
+- Less advanced SQL and JSON functionality compared to PostgreSQL.
+- Historically more limited full-text search and window function support.
+
+## Decision
+
+We chose **PostgreSQL** as the primary database for storing user subscriptions.
```

Comment:
The decisions lack a description explaining why this particular technology was chosen.